schemas/pipeline_test.go

```diff
@@ -0,0 +1,229 @@
+package schemas
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestPipelineValidate(t *testing.T) {
+	t.Run("no errands", func(t *testing.T) {
+		p := Pipeline{Name: "no errands!"}
+		assert.EqualError(t, p.Validate(), "no errands specified in pipeline")
```

Comment:
is this error a sentinel value anywhere?

Reply:
No, I just wanted to make sure the validation is failing for the right reasons, but I didn't really think it was worth creating named errors for all of the different cases

Reply:
yeah.. just feels like it is too sensitive of a test and will get annoying if we change the language of the error or do want to change it in a different way. Ultimately we don't care if it fails the exact way we want as long as it fails. I view it more as a truth table where it either fails or not.

Reply:
It is a minor annoyance, but I think it does add value by asserting particular errors. It was very useful while I was writing the function and helped me catch some mistakes. Since we can never be exhaustive with test cases here, seeing that an error message changed completely should be an indicator to double check nothing went wrong